qutip/solver/heom/bofin_baths.py

```diff
@@ -973,10 +1057,621 @@ def _calc_eps(self, Nk):
 
     def _calc_chi(self, Nk):
         alpha_p = np.diag([
-                1. / np.sqrt((2 * k + 5) * (2 * k + 3))
-                for k in range(2 * Nk - 2)
+            1. / np.sqrt((2 * k + 5) * (2 * k + 3))
+            for k in range(2 * Nk - 2)
         ], k=1)
         alpha_p += alpha_p.transpose()
         evals = eigvalsh(alpha_p)
         chi = [-2. / val for val in evals[0: Nk - 1]]
         return chi
+
+
+class FitSpectral(BosonicBath):
+    """
+    A helper class for constructing a Bosonic bath from a spectral density fit.
+
+    Parameters
+    ----------
+    Q : Qobj
+        Operator describing the coupling between system and bath.
+
+    T : float
+        Bath temperature.
+
+    Nk : int
+        Number of exponential terms used to approximate the bath correlation
+        functions.
+    """
+
+    def __init__(self, T, Q, Nk=14):
+        self.Q = Q
+        self.T = T
+        self.beta = 1 / self.T
+        self.Nk = Nk
+
+    def __str__(self):
+        try:
+            lam, gamma, w0 = self.params_spec
+            summary = gen_summary(self.fit_time, self._rmse, self.spec_n,
+                                  "The Spectral Density", lam, gamma, w0)
+            return summary
+        except NameError:
+            return "Fit correlation instance: \n No fit has been performed yet"
+
+    def summary(self):
+        print(self.__str__())
+
+    def spectral_density_approx(self, w, a, b, c):
```

Comment:
I did rename spectral_density_approx -> _spectral_density_approx , but kept spectral_density -> spectral_density_approx , mainly because if I keep the name spectral_density then I can use the power_spectrum method that I put in BosonicBath, of course maybe all these methods will be moved around, but I guess I'll wait on the highlevel comments before renaming this one